### ISSUE TITLE
FirestoreDbBuilder.ConverterRegistry ignored when using Firestore Emulator

### ISSUE THREAD

**Issue by jrojasledesma** [googleapis/google-cloud-dotnet#6313]
Hi. I'm having problems trying to use custom type converters via the Converter Registry in FirestoreDbBuilder when targeting the Firestore Emulator.

Looking a the source code I noticed that FirestoreDbBuilder ignores the value of its ConverterRegistry property when creating an instance of FirestoreDb. The problem seems to be in lines 186-192 of FirestoreDbBuilder.cs, inside the function `MaybeUseEmulator`:

```
...
 return new FirestoreDbBuilder
{
    Endpoint = hostAndPort,
    Settings = settings,
    ProjectId = ProjectId,
    ChannelCredentials = ChannelCredentials.Insecure
};
```
Note that in the new FirestoreDbBuilder returned by this method, ConverterRegistry is not asigned. Thus, the original value is ignored when the FirestoreDb instance is created in the recursive call to `Build` in line 89 of FirestoreDbBuilder.cs.

Is there any good reason to ignore the value, or is this simply a bug?

**Comment by jskeet** [comment]
I'll have a look when I'm back at work tomorrow. It's almost certainly just a bug.

**Comment by jskeet** [comment]
(I suspect it will be a bug that's easily fixed, and which we can then just roll out a patch release for. But we'll see...)

**Comment by jrojasledesma** [comment]
Thanks! Good to know this could be fixed  soon.

**Comment by jskeet** [comment]
Hi @jrojasledesma, the fix is now released in version 2.3.1. I'm pretty confident in the fix, but please shout if you have any problems :)

**Comment by jrojasledesma** [comment]
Hi, I just tried the new version and confirmed that my issue is now fixed. Thank you @jskeet and @amanda-tarafa!
